persistence/src/main/scala/za/co/absa/spline/persistence/ArangoDatabaseFacade.scala

```diff
@@ -21,12 +21,38 @@ import com.arangodb.async.{ArangoDBAsync, ArangoDatabaseAsync}
 import org.springframework.beans.factory.DisposableBean
 import za.co.absa.commons.lang.OptionImplicits.AnyWrapper
 
+import javax.net.ssl._
+import java.security.SecureRandom
+import java.security.cert.X509Certificate
+
 class ArangoDatabaseFacade(connectionURL: ArangoConnectionURL) extends DisposableBean {
 
-  private val ArangoConnectionURL(maybeUser, maybePassword, host, port, dbName) = connectionURL
+  private val ArangoConnectionURL(_, maybeUser, maybePassword, host, port, dbName) = connectionURL
+
+  private val isSecure = connectionURL.isSecure
+
+  private def createSslContext: SSLContext = {
+    val sslContext = SSLContext.getInstance("SSL")
+    sslContext.init(null, Array(TrustAll), new SecureRandom())
+    sslContext
+  }
+
+  // Bypasses both client and server validation.
+  private object TrustAll extends X509TrustManager {
+    val getAcceptedIssuers = null
+    def checkClientTrusted(x509Certificates: Array[X509Certificate], s: String) = {}
+    def checkServerTrusted(x509Certificates: Array[X509Certificate], s: String) = {}
+  }
+
+  // Verifies all host names by simply returning true.
+  private object VerifiesAllHostNames extends HostnameVerifier {
```

Comment:
Is `VerifiesAllHostNames` used anywhere? What was the intention?

Reply:
I am honest this is a SO rip, trying to implement unsecure SSL with basic representations.